### PR TITLE
Document use of SPIRV-Tools with asm output

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
Add information into readme about the use of the translator with assembly from SPIRV-Tools.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+from a custom location using the following instructions:
+
+1. Checkout, build and install SPIRV-Tools using
+   [the following instructions](https://github.com/KhronosGroup/SPIRV-Tools#build).
+   Example using cmake:
+```
+cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=<location to installation>
+ninja install
+```
+2. Using pkg-config when building the translator by setting
```
Comment (AnastasiaStulova) [inline README.md:110]:
```suggestion
2. Use pkg-config when building the translator by setting
```

File: README.md [hunk 2]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
   moment, see [KhronosGroup/SPIRV-LLVM-Translator#477](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/477)
   to track progress, discuss and contribute.
 
+### Build with SPIRV-Tools
+
+The translator can use SPIRV-Tools to generate assembly with conformant syntax.
```
Comment (AlexeySotkin) [inline README.md:98]: It would be nice to have a link to the SPIR-V Tools repo.
```suggestion
The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with conformant syntax.
```

File: README.md [hunk 3]
```diff
@@ -153,6 +181,7 @@ To translate between LLVM IR and SPIR-V:
     * `-o file_name` - to specify output name
     * `-spirv-debug` - output debugging information
     * `-spirv-text` - read/write SPIR-V in an internal textual format for debugging purpose. The textual format is not defined by SPIR-V spec.
+    * `--spirv-tools-dis` - print SPIR-V assembly in standard format. Only available on [builds with SPIR-V](#build-with-spirv-tools).
```
Comment (AlexeySotkin) [inline README.md:184]: I don't think there is a really _standard_ format for textual representation of SPIR-V.

Also a typo in the link
```suggestion
    * `--spirv-tools-dis` - print SPIR-V assembly in standard format. Only available on [builds with SPIRV-Tools](#build-with-spirv-tools).
```

File: README.md [hunk 4]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+```
+2. Using pkg-config when building the translator by setting
+   `PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/` variable
+   before the cmake line invocation.
+   Example:
+```
+PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/ cmake ...
+```
+
+To test the correctness of translator build run the following line
```
Comment (svenvh) [inline README.md:118]:
```suggestion
To verify the SPIR-V Tools integration in the translator build, run the following line
```

File: README.md (hunk 1 above)
Comment (AnastasiaStulova) [reply README.md:110]: I am not sure what part is not correct - it is saying that by means of pkg-config we can specify the SPIRV-Tools installation...

File: README.md (hunk 3 above)
Comment (AnastasiaStulova) [reply README.md:184]: Hmm, although I thought that SPIRV-Tools produce the standard syntax. I am using this terminology in clang docs too. I think it will be more confusing if we start talking about SPIRV-Tools syntax because then it is not clear why would anyone bother about some syntax from another project. We should bear in mind that this documentation is not for top experts in the tooling aspects of this area.

File: README.md [hunk 5]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+The translator can use SPIRV-Tools to generate assembly with conformant syntax.
+If SPIRV-Tools have been installed prior to the build it will be detected and
+used automatically. However it is also possible to enable use of SPIRV-Tools
+from a custom location using the following instructions:
+
+1. Checkout, build and install SPIRV-Tools using
+   [the following instructions](https://github.com/KhronosGroup/SPIRV-Tools#build).
+   Example using cmake:
+```
+cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=<location to installation>
```
Comment (AnastasiaStulova) [inline README.md:107]:
```suggestion
cmake [-G <platform-generator>] <SPIRV-Tools location> -DCMAKE_INSTALL_PREFIX=<installation location>
```

File: README.md [hunk 6]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+ninja install
+```
+2. Using pkg-config when building the translator by setting
+   `PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/` variable
+   before the cmake line invocation.
+   Example:
+```
+PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/ cmake ...
```
Comment (AnastasiaStulova) [inline README.md:115]:
```suggestion
PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/ cmake <other options>
```

File: README.md [hunk 7]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+### Build with SPIRV-Tools
+
+The translator can use SPIRV-Tools to generate assembly with conformant syntax.
+If SPIRV-Tools have been installed prior to the build it will be detected and
+used automatically. However it is also possible to enable use of SPIRV-Tools
+from a custom location using the following instructions:
+
+1. Checkout, build and install SPIRV-Tools using
+   [the following instructions](https://github.com/KhronosGroup/SPIRV-Tools#build).
+   Example using cmake:
```
Comment (AnastasiaStulova) [inline README.md:105]:
```suggestion
   Example using CMake with Ninja:
```

File: README.md (hunk 1 above)
Comment (svenvh) [reply README.md:110]: Maybe rephrase it as:

```suggestion
2. Point pkg-config to the SPIR-V tools installation when configuring the translator by setting
```

File: README.md [hunk 8]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
...
+```
+cmake -G Ninja .. -DCMAKE_INSTALL_PREFIX=<location to installation>
+ninja install
+```
+2. Using pkg-config when building the translator by setting
+   `PKG_CONFIG_PATH=<location to installation>/lib/pkgconfig/` variable
```
Comment (AnastasiaStulova) [inline README.md:111]:
```suggestion
   `PKG_CONFIG_PATH=<SPIRV-Tools installation location>/lib/pkgconfig/` variable
```

File: README.md (hunk 3 above)
Comment (AnastasiaStulova) [reply README.md:184]: ok, I had the impression that SPIRV-Tools produced a defacto standard syntax because it is used in the examples in the spec... but if we don't align on this then I would say we should not write overstatements in the public docs.

Let's go for` SPIRV-Tools syntax` for now even though I don't like exposing the implementation details to the users... but at least it is not an incorrect statement.

File: README.md [hunk 9]
```diff
@@ -93,6 +93,34 @@ Building clang from sources takes time and resources and it can be avoided:
   moment, see [KhronosGroup/SPIRV-LLVM-Translator#477](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/issues/477)
   to track progress, discuss and contribute.
 
+### Build with SPIRV-Tools
+
+The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with conformant syntax.
```
Comment (AnastasiaStulova) [inline README.md:98]:
```suggestion
The translator can use [SPIRV-Tools](https://github.com/KhronosGroup/SPIRV-Tools) to generate assembly with widely adopted syntax.
```

**Review by svenvh — APPROVED** [review]
